### PR TITLE
Add strings

### LINKED ISSUE #25: Add strings

### PR DESCRIPTION
**Pull request by jamespeilunli**
closes #25

### CODE REVIEW

**Comment by jamespeilunli** [comment]
In functions, argument names surrounded by spaces inside strings will be replaced by their values. We need to update the SourceCode::replace_args function to fix this.

**Comment by jamespeilunli** [comment]
We've decided to make a separate PR to fix the bug and push this PR.

File: src/main.cpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #include "../inc/source_code.hpp"
+#include "../inc/type.hpp"
 #include "../inc/parser.hpp"
```
Comment (spierceVR) [inline src/main.cpp]:
```suggestion
#include "source_code.hpp"
#include "type.hpp"
#include "parser.hpp"
```

CMake's `include_directories()` takes care of this for us https://cmake.org/cmake/help/latest/command/include_directories.html

**Review by spierceVR — APPROVED** [review]
LGTM

**Pull request merged**